File: alice/divkt_renderer/src/main/java/projects/example/ExampleDiv.kt
package ru.yandex.alice.divktrenderer.projects.example

import com.yandex.div.dsl.context.*
import com.yandex.div.dsl.model.*
import com.yandex.div.dsl.reference
import com.yandex.div.dsl.type.color
import com.yandex.div.dsl.value
import ru.yandex.alice.divkit.IdempotentDivKitTemplate
import ru.yandex.alice.divkit.StaticContext
import ru.yandex.alice.protos.data.scenario.example.Data.TExampleScenarioData
import kotlin.reflect.KClass

class ExampleDiv() : IdempotentDivKitTemplate<TExampleScenarioData> {
    override val templateDataClass: KClass<TExampleScenarioData>
        get() = TExampleScenarioData::class

    private val greetingsTextRef = reference<String>("greetings_text")

    override fun render(ctx: StaticContext, data: TExampleScenarioData): CardWithTemplates {
        val templates: TemplateContext<Div> = templates {
            define(
                "greetings",
                divText(
                    text = greetingsTextRef,
                    textColor = value("#ff0".color)
                )
            )
        }
        val card = card {
            divData(
                logId = "example_card",
                states = listOf(
                    state(
                        stateId = 0,
                        div = divContainer(
                            items = listOf(
                                greetings(data)
                            )
                        )
                    )
                )
            )
        }
        return CardWithTemplates(card, templates)
    }

    private fun CardContext.greetings(data: TExampleScenarioData) = template(
        "greetings",
        resolve(greetingsTextRef, data.hello ?: " ")
    )
}

File: alice/div-renderer/src/projects/example/templates/ExampleDiv/ExampleDiv.ts
import { ContainerBlock } from 'divcard2';
import { NAlice } from '../../../../protos';
import { TopLevelCard } from '../../../centaur/helpers/helpers';
import { MMRequest } from '../../../../common/helpers/MMRequest';
import { IRequestState } from '../../../../common/types/common';
import Greetings from '../../components/Greetings/Greetings';
type ExampleData = NAlice.NData.ITExampleScenarioData;

export default function ExampleDiv({ hello }: ExampleData, _: MMRequest, requestState: IRequestState) {
    return TopLevelCard({
        log_id: 'example_card',
        states: [
            {
                state_id: 0,
                div: new ContainerBlock({
                    items: [
                        Greetings({
                            hello: hello ?? ' ',
                        }, requestState),
                    ],
                }),
            },
        ],
    }, requestState);
}
